listeners.js

```diff
@@ -0,0 +1,156 @@
+const _ = require('lodash')
+const jsonpointer = require('jsonpointer')
+
+class Listener{
+    constructor(schema, ui_schema, formRef){
+        this.schema = Object.assign({}, schema)
+        this.ui_schema = Object.assign({}, ui_schema)
+        this.formRef = formRef
+        this.listeners = this._getListeners()
+    }
+
+    _getListeners() {
+        let acc = {}
+        this._getListenersFromUISchema(this.ui_schema, '', acc)
+        return acc
+    }
+
+    _getListenersFromUISchema(ui_schema, path, acc) {
+        const listeners = ui_schema["listener"] || ui_schema["logic:listener"]
+        if(listeners){
+            acc[path] = (acc[path] || [])
+            if (_.isArray(listeners)) {
```

Comment:
Can we start to move this away from `lodash` as much as possible? `Array.isArray` here, `forEach` to replace `_each`, etc.

Reply:
I'm fine with using forEach but why the hesitation of using lodash? The type checking interface is atrocious in native js. No difference between number/integer, can't check for array with typeof, you end up having to rewrite what lodash does with it's consistent interface so why reinvent the wheel?

Reply:
It's more that if we can narrow down to just the parts of lodash we actually need we can reduce the footprint by only importing those.